scalafix-core/src/main/scala/scalafix/internal/util/PrettyType.scala

```diff
@@ -301,35 +311,44 @@ class PrettyType private (
                   Nil,
                   inits,
                   Self(Name(""), None),
-                  objectDecls
+                  objectDecls,
+                  Nil
                 )
               )
             case k.CLASS =>
               val ctor: Ctor.Primary = declarations
                 .collectFirst {
                   case i if i.kind.isConstructor && i.is(p.PRIMARY) =>
                     toTree(i) match {
-                      case ctor @ Ctor.Primary(_, _, Nil :: Nil)
-                          if !info.is(p.CASE) =>
+                      case ctor @ Ctor.Primary.After_4_6_0(
+                            _,
+                            _,
+                            Term.ParamClause(Nil, _) :: Nil
+                          ) if !info.is(p.CASE) =>
                         // Remove redudant () for non-case classes: class Foo
                         ctor.copy(paramss = Nil)
                       case e: Ctor.Primary => e
                     }
                 }
                 .getOrElse {
-                  Ctor.Primary(Nil, Name(""), Seq.empty[Term.ParamClause])
+                  Ctor.Primary.After_4_6_0(
```

Comment:
i thought specific versions were only needed for unapply

Reply:
Good catch, I didn't even remember an apply was created for the specific versions. I guess I just changed that because of a bad search, but there is no reason to. In any case, this PR needs so much rebase that I would (will?) probably start from scratch.